base/src/main/java/org/aya/distill/CoreDistiller.java

```diff
@@ -159,6 +159,17 @@ yield visitCalls(true, defVar(var),
           ? linkLit(0, zero.ref, CON_CALL)
           : linkLit(shaped.repr(), suc.ref, CON_CALL),
         () -> Doc.plain(String.valueOf(shaped.repr())));
+      case LitTerm.ShapedList shaped -> {
+        var display = Doc.sep(
+          Doc.symbol("["),
+          Doc.commaList(shaped.repr().map(x -> term(Outer.Free, x))),
+          Doc.symbol("]"));
+
+        yield shaped.with((nil, cons, dataArg) -> {
+          // link to data, i think.
+          return linkListLit(display, nil.dataRef, DATA_CALL);   // Maybe we should use other style.
```

Comment:
Maybe we can link `[` to `cons` and `]` to `nil`, which looks funny!

Reply:
How about linking `,` to `cons`, `[` and `]` to `nil` ?

Reply:
> How about linking `,` to `cons`, `[` and `]` to `nil` ?

I _love_ this idea